pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap_sysprep.go

```diff
@@ -79,3 +84,74 @@ func BootstrapSysPrep(
 
 	return configSpec, customSpec, nil
 }
+
+func convertTo(from *vmopv1_prep.Sysprep, bootstrapData BootstrapData) *vimTypes.CustomizationSysprep {
+	sysprepCustomization := &vimTypes.CustomizationSysprep{}
+
+	if from.GUIUnattended != nil {
+		sysprepCustomization.GuiUnattended = vimTypes.CustomizationGuiUnattended{
+			Password: &vimTypes.CustomizationPassword{
+				Value:     *bootstrapData.SysprepPassword,
```

Comment:
Will this be `nil` when `GUIUnattended.AutoLogon` false?

Reply:
Removed the pointer to avoid an accidental NPE, also moved the logic to add it only if the value is set in `bootstrapData`.